src/Dialogs/FrequencyDialog.cpp

```diff
@@ -167,24 +179,18 @@ FrequencyListWidget::UpdateList() noexcept
 void
 FrequencyDialogShowModal() noexcept
 {
-  static std::vector<FrequencyListRenderer::RadioChannel> channels;
+  static std::vector<FrequencyListWidget::RadioChannel> channels;
 
   const DialogLook &look = UIGlobals::GetDialogLook();
 
   auto widget = std::make_unique<FrequencyListWidget>(look,&channels);
 
-  static bool first = true;
-  static bool haveRows = false;
-
-  if (first == true || FrequenciesFileChanged == true) {
+  if (channels.size() == 0 || FrequenciesFileChanged == true) {
```

Comment:
Never check `size()==0`, try `empty()` instead, which is often faster.
What happens if the file is empty? Will it reload the file every time?